### ISSUE TITLE
`Style/IfUnlessModifer` combined with `Style/HashSytax: EnforcedShorthandSyntax` results in `unexpected token $end` error

### ISSUE THREAD

**Issue by madhermit** [rubocop/rubocop#11182]
Given a particular block of code, the `Style/IfUnlessModifer` cop, combined with the `Style/HashSytax: EnforcedShorthandSyntax` cop results in a `Lint/Syntax: unexpected token $end` error.

Given the following ruby code:
```ruby
if foo
  json.partial! 'foo', foo: foo
end
```
Running `rubocop -a` with a default configuration, results in the following:
```ruby
json.partial! 'foo', foo: if foo
```
--------

## Expected behavior

I expected RuboCop to not replace code in such a way that it results in an error. A solution to this would be to wrap the hash parameters in parenthesis. For example:
```ruby
json.partial!('foo', foo:) if foo
```

## Actual behavior

Results in:
```ruby
json.partial! 'foo', foo: if foo
```
Which results in the following error: `rubocop: unexpected token $end`

## Steps to reproduce the problem

Use the following ruby code:
```ruby
if foo
  json.partial! 'foo', foo: foo
end
```
Then run `rubocop -a` with the default configuration.

## RuboCop output

```
$ [bundle exec] rubocop -a --debug

Inspecting 1 file
E

Offenses:

rubocopissue.rb:1:1: C: [Corrected] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
if foo
^^
rubocopissue.rb:2:1: E: Lint/Syntax: unexpected token $end
(Using Ruby 3.1 parser; configure using TargetRubyVersion parameter, under AllCops)
rubocopissue.rb:2:29: C: [Corrected] Style/HashSyntax: Omit the hash value.
  json.partial! 'foo', foo: foo
                            ^^^

1 file inspected, 3 offenses detected, 2 offenses corrected

```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.38.0 (using Parser 3.1.2.1, rubocop-ast 1.23.0, running on ruby 3.1.0) [x86_64-linux]
```
